test/include/seqan3/test/performance/sequence_generator.hpp

```diff
@@ -26,34 +27,83 @@
+//!\cond
+    requires (std::ranges::output_range<sequence_t, std::ranges::range_value_t<sequence_t>>)
+//!\endcond
+class random_sequence_generator
 {
-    std::mt19937 gen(seed);
-    size_t max_val{};
-
-    if constexpr (std::unsigned_integral<alphabet_t>)
-        max_val = std::numeric_limits<size_t>::max();
-    else
-        max_val = alphabet_size<alphabet_t> - 1ull;
-
-    std::uniform_int_distribution<size_t> dis_alpha(0ull, max_val);
-    std::uniform_int_distribution<size_t> dis_length(len - variance, len + variance);
+public:
+
+    //!\brief Stores the size of the random sequence.
+    size_t size{};
+    //!\brief Stores the variance of the random sequence.
+    size_t size_variance{};
+
+    random_sequence_generator() = default; //!< Defaulted
+    random_sequence_generator(random_sequence_generator const &) = default; //!< Defaulted
+    random_sequence_generator(random_sequence_generator &&) = default; //!< Defaulted
+    random_sequence_generator & operator=(random_sequence_generator const &) = default; //!< Defaulted
+    random_sequence_generator & operator=(random_sequence_generator &&) = default; //!< Defaulted
+    ~random_sequence_generator() = default; //!< Defaulted
+
+    /*!\brief Initialises a random sequence generator which generates sequences with a given mean size.
+     * \param[in] size          The size of the random sequence.
+     * \param[in] size_variance The variance of the random sequence size.
+     */
+    explicit random_sequence_generator(size_t size, size_t size_variance = 0) :
+        size{size},
+        size_variance{size_variance}
+    {}
+
+    /*!\name Element access
+     * \{
+     */
+
+    /*!\brief Returns a random sequence for a set size (and variance).
+     * \param[in,out] random_generator e.g. std::mt19937 with a seed of 42
+     * \returns a random generated sequence.
+     */
+    template <typename generator_t>
+    sequence_t operator()(generator_t && random_generator) const
+    {
+        using alphabet_t = std::ranges::range_value_t<sequence_t>;
 
-    std::vector<alphabet_t> sequence;
+        size_t max_alphabet_rank{};
 
-    size_t length = dis_length(gen);
-    for (size_t l = 0; l < length; ++l)
-    {
-        if constexpr (std::unsigned_integral<alphabet_t>)
-            sequence.push_back(dis_alpha(gen));
+        if constexpr (std::is_same_v<alphabet_t, uint64_t>)
+            max_alphabet_rank = std::numeric_limits<size_t>::max();
         else
-            sequence.push_back(assign_rank_to(dis_alpha(gen), alphabet_t{}));
+            max_alphabet_rank = alphabet_size<alphabet_t> - 1ull;
+
+        std::uniform_int_distribution<size_t> alphabet_rank_distribution(0ull, max_alphabet_rank);
+        std::uniform_int_distribution<size_t> sequence_size_distribution(size - size_variance, size + size_variance);
+
+        size_t sequence_size = sequence_size_distribution(random_generator);
+        sequence_t random_sequence(sequence_size);
```

Comment:
Pheww, can we actually assume that this does in fact always does what we expect? I think this is not handled properly by the concepts yet. The biggest problem that I see is handling the SeqAn2 sequence types which do not have push_back or resize member functions. So I guess later on we need to use a helper function to differentiate between sequences that have a resize member function. For now I propose to add this inline requires clause in the head of the class template: 

```cpp
requires requires (sequence_t & s) { {s.resize()}; }
```

And then use here the resize member function instead of the constructor.